pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -606,15 +598,10 @@ public static void addColumnMetadataInfo(PropertiesConfiguration properties, Str
       }
     }
 
-    // datetime field
-    if (fieldSpec.getFieldType().equals(FieldType.DATE_TIME)) {
-      DateTimeFieldSpec dateTimeFieldSpec = (DateTimeFieldSpec) fieldSpec;
-      properties.setProperty(getKeyFor(column, DATETIME_FORMAT), dateTimeFieldSpec.getFormat());
-      properties.setProperty(getKeyFor(column, DATETIME_GRANULARITY), dateTimeFieldSpec.getGranularity());
-    }
-
+    addFieldSpec(properties, column, fieldSpec);
+    DataType dataType = fieldSpec.getDataType();
     // NOTE: Min/max could be null for real-time aggregate metrics.
-    if (totalDocs > 0) {
+    if ((fieldSpec.getFieldType() != FieldType.COMPLEX) && (totalDocs > 0)) {
```

Comment:
The refactor code here changes the ordering of each column props, which changes the CRC. 
@Jackie-Jiang